src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +238,287 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
+                    } break;
+                    case SECURITY_ANALYSIS: {
+                        networkModificationTreeService.updateSecurityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SECURITY_ANALYSIS_STATUS;
+                    } break;
+                    case SENSITIVITY_ANALYSIS: {
+                        networkModificationTreeService.updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_STATUS;
+                    } break;
+                    case SHORT_CIRCUIT: {
+                        // TODO : this Stopped function is the only one when SHORT_CIRCUIT doesn't handle buses => might cause a bug
+                        networkModificationTreeService.updateShortCircuitAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_SHORT_CIRCUIT_STATUS;
+                    } break;
+                    case VOLTAGE_INITIALIZATION: {
+                        networkModificationTreeService.updateVoltageInitResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_VOLTAGE_INIT_STATUS;
+                    } break;
+                    case DYNAMIC_SIMULATION: {
+                        networkModificationTreeService.updateDynamicSimulationResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_DYNAMIC_SIMULATION_STATUS;
+                    } break;
+                }
 
-                    // delete sensitivity analysis result in database
-                    updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
+                // send notification for stopped computation
+                notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), updateType);
+            } catch (JsonProcessingException e) {
+                LOGGER.error(e.toString());
+            }
+        }
+    }
 
-                    // send notification for stopped computation
-                    UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
-                    notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), NotificationService.UPDATE_TYPE_SENSITIVITY_ANALYSIS_STATUS);
-                } catch (JsonProcessingException e) {
-                    LOGGER.error(e.toString());
+    public void consumeCalculationResult(Message<String> msg, ComputationType computationType) {
+        UUID resultUuid = null;
+        String resultId = msg.getHeaders().get(RESULT_UUID, String.class);
+        if (resultId != null) {
```

Comment:
resultId can be null ? i don't see this test in the old  code.

Reply:
It was only in one of the code version that I reformated. I kept it but it looks very unlikely.